CMakeLists.txt

```diff
@@ -19,37 +23,60 @@ find_package(tf2_geometry_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(Eigen3 REQUIRED)
 
+set(includes
```

Comment:
This isn't necessary anymore, use `ament_target_dependencies` see examples here https://github.com/ros-planning/navigation2/blob/master/nav2_planner/CMakeLists.txt

Reply:
Hm I think I might have missed something, because couldn't get it to find some includes (eigen3 and diagnostics_updated without adding them). 

See my changes, I think I need some help on that one.

Reply:
Sure let me know

Reply:
Of everything I'm the least concerned by the CMake stuff, give it another shot and if not I can take care of it for you after the fact